otter/cloud_client.py

```diff
@@ -169,10 +212,12 @@ def got_auth((token, catalog)):
         else:
             request_ = add_bind_service(
                 catalog, service_name, region, log, request_)
-        if service_request.json_response:
-            request_ = add_json_response(request_)
         request_ = add_error_handling(
             service_request.success_pred, request_)
+        if service_request.json_response:
+            request_ = add_json_response(request_)
```

Comment:
Hm... I changed the CLB ones which expect JSON to expect text.  I understand what you're saying though, although as an amendment to your proposal I think it should raise an error and return text, maybe?  Otherwise the callers never know whether they're getting text or JSON, and will have to test every time.